File: linkedinsdk/src/main/java/com/AbdAllahAbdElFattah13/linkedinsdk/linkedin_builder/LinkedInFromFragmentBuilder.java
package com.AbdAllahAbdElFattah13.linkedinsdk.linkedin_builder;

import android.content.Context;
import android.content.Intent;

import androidx.fragment.app.Fragment;

public final class LinkedInFromFragmentBuilder extends LinkedInBuilder {

    private Fragment fragment;

    private LinkedInFromFragmentBuilder(Fragment fragment) {
        this.fragment = fragment;
    }

    public static LinkedInFromFragmentBuilder getInstance(Fragment fragment) {
        return new LinkedInFromFragmentBuilder(fragment);
    }

    @Override
    Context getContext() {
        return this.fragment.getContext();
    }

    @Override
    LinkedInBuilder self() {
        return this;
    }

    @Override
    void startActivityForResult(Intent intent, int requestCode) {
        fragment.startActivityForResult(intent, requestCode);
    }
}

File: linkedinsdk/src/main/java/com/AbdAllahAbdElFattah13/linkedinsdk/ui/linkedin_builder/LinkedInFromFragmentBuilder.kt
package com.AbdAllahAbdElFattah13.linkedinsdk.ui.linkedin_builder

import android.content.Intent
import androidx.fragment.app.Fragment

class LinkedInFromFragmentBuilder private constructor(private val fragment: Fragment) : LinkedInBuilder(fragment.activity) {
    override fun self(): LinkedInBuilder {
        return this
    }

    override fun startActivityForResult(intent: Intent, requestCode: Int) {
        fragment.startActivityForResult(intent, requestCode)
    }

    companion object {
        fun getInstance(fragment: Fragment): LinkedInFromFragmentBuilder {
            return LinkedInFromFragmentBuilder(fragment)
        }
    }
}